mozilla-release/browser/components/urlbar/UrlbarInput.jsm

```diff
@@ -1090,7 +1090,19 @@ class UrlbarInput {
 
     this.valueIsTyped = false;
     this._resultForCurrentValue = null;
-    this.inputField.value = val;
+
+    // DB-2378 => CLIQZ-SPECIAL: do not set url if freshtabURL exists,
+    // do not check if custom URL is set by user
+    const homeUrl = Services.prefs.getCharPref("browser.startup.homepage", "");
```

Comment:
What about reading the pref only if `val.startsWith("moz-ext")` (which I assume will be false in more than 99% of cases). Then there would be no overhead at all?

Reply:
then we will need to add one more condition

Reply:
What about:
```js
    // DB-2378 => CLIQZ-SPECIAL: do not set url if freshtabURL exists,
    // do not check if custom URL is set by user
    if (
      val.startsWith("moz-ext") &&
      val === Services.prefs.getCharPref("browser.startup.homepage", "")
    ) {
      this.inputField.value = "";
    } else {
      this.inputField.value = val;
    }
```

Reply:
I suggested:

Reply:
this.inputField.value = val;
    if (val.startsWith("moz-ext")) {
      const homeUrl = Services.prefs.getCharPref("browser.startup.homepage", "");
      if (val === homeUrl) {
        this.inputField.value = "";
      }
    }

Reply:
Up to you guys!

Reply:
Good idea anyway, we will fix this